src/wazuh_modules/wm_database.c

```diff
@@ -460,38 +459,11 @@ int wm_sync_file(const char *dirname, const char *fname) {
         return -1;
     }
 
-    switch (type) {
-
-    case WDB_GROUPS:
-        id_agent = atoi(fname);
-
-        if (!id_agent) {
-            mterror(WM_DATABASE_LOGTAG, "Couldn't extract agent ID from file %s/%s", dirname, fname);
-            return -1;
-        }
-
-        if (wdb_get_agent_status(id_agent, &wdb_wmdb_sock) < 0) {
-            snprintf(del_path, PATH_MAX - 1, DEFAULTDIR GROUPS_DIR "/%03d", id_agent);
-            unlink(del_path);
-            wdb_delete_agent_belongs(id_agent, &wdb_wmdb_sock);
-            return -1;
-        }
+    id_agent = atoi(fname);
```

Comment:
I think this should be done only for the `WDB_GROUPS` case. Shared files don't have a number as a name and they will fail in the if below.

Reply:
Fixed